.github/workflows/build.yml

```diff
@@ -143,6 +143,11 @@ jobs:
           make regen-deepfreeze
           make -j4 regen-all
           make regen-stdlib-module-names
+      - name: Run patchcheck
+        if: github.event_name == 'pull_request'
+        run: |
+          git fetch origin
```

Comment:
Do we need to fetch origin? It takes 1m 49s for this step.

We don't do it on [Azure Pipelines and `patchcheck` takes 2s](https://dev.azure.com/Python/cpython/_build/results?buildId=137844&view=logs&j=256d7e09-002a-52d7-8661-29ee3960640e&t=3d7276d3-4e8d-5309-55ad-fb0b172d9925).

Reply:
Let's try :)

I think that we might need it because of the heavy `git` machinery inside `patchcheck`.
I think we might need it during backports for older branches.

Reply:
Nope, it does not work:

```
Run # git fetch origin
Checked 107 modules (31 built-in, 75 shared, 1 n/a on linux-x86_64, 0 disabled, 0 missing, 0 failed on import)
LD_LIBRARY_PATH=/home/runner/work/cpython/cpython ./python ./Tools/patchcheck/patchcheck.py --ci true
Getting base branch for PR ... origin/main
fatal: ambiguous argument 'origin/main': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
error running git diff --name-status origin/main
make: *** [Makefile:2914: patchcheck] Error 1
Getting the list of files that have been added/changed ... 
```